networkx/lazy_imports.py

```diff
@@ -81,6 +81,18 @@ def __dir__():
     return __getattr__, __dir__, list(__all__)
 
 
+class DelayedImportErrorModule(types.ModuleType):
+    def __init__(self, import_name, *args, **kwargs):
+        self.__import_name = import_name
+        super().__init__(*args, **kwargs)
+
+    def __getattr__(self, x):
+        if x.startswith("__"):
```

Comment:
The impact is that, if a user tries to access `scipy.__some_developer_attribute` it will break.  It may be worth limiting this to specific methods, but I wasn't sure which ones are sufficient to, e.g., print the module and use it in the ways you normally would.  If you say the previous code *only* had access to `__class__` and `__spec__` I can restore similar behavior.

Reply:
I've now limited this to `__class__`, `__file__`, and `__import_name` (our variable).  That way, you can at least print the module without trouble.

Reply:
Hmmmm...   not sure this works for printing. What should it print?
```
>>> np = nx.lazy_imports.lazy_import('numpy')
>>> print(np)
<module 'DelayedImportErrorModule'>
>>> print(np.__file__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dschult/NXstefanv/networkx/lazy_imports.py", line 91, in __getattr__
    super().__getattr__(x)
AttributeError: 'super' object has no attribute '__getattr__'
```

Reply:
Yes, the `__file__` attribute is necessary to make `print(np)` work.